Ties the new lawyer speech bubble to the attorney's badge

Incoming5643 opened the pull request:
So lawyers can disguise as non-lawyers and visa versa

Additional/replacement badges are found in the law closet

Sprite by @WJohn <img src=https://camo.githubusercontent.com/e168933eb88ce377bd0bfe18d955345ce8bfa493/687474703a2f2f692e696d6775722e636f6d2f564948336d6c412e706e67>

Incoming5643 committed:
ties the lawyer speech bubble to the attorney's badge so lawyers can disguise and people can disguise as lawyers

additional badges are found in the law closet

Needs an icon

WJohn commented:
God that sprite is bad let me make one for you:
![](http://i.imgur.com/tURbxpt.png)

PKPenguin321 commented:
wj comes through

lordpidey commented on code/modules/clothing/under/ties.dm:
```diff
@@ -367,3 +367,22 @@
+/obj/item/clothing/tie/attach(obj/item/clothing/under/U, user)
+	if(!..())
+		return 0
+	user.bubble_icon = "lawyer"
```
Uhh, couple of problems I notice about this.

1.) You need to cast and check if user is null.
2.) Won't this change the voice bubble of the person who pins, not the person who is pinned (usually the same person, but not always)
3.) What if the shirt is pinned while not equipped?

Incoming5643 replied:
There's work to be done yes, I just lost hope until a sprite came throguh. I'll now go finish the pull.

optimumtact replied:
if user is null at this point then something very bad has happened, as there are no sleeps between this and the parent call, which should be checking this

Incoming5643 replied:
this code is already gone, don't panic about it.

Thanks github for showing it anyways,

Incoming5643 committed:
finishes the job (thanks WJ)

RemieRichards commented on code/modules/clothing/under/ties.dm:
```diff
@@ -367,3 +367,37 @@
+		current_lawyer.bubble_icon = initial(current_lawyer.bubble_icon)
+		current_lawyer = null
+
+/obj/item/clothing/tie/lawyers_badge/process()
```
No, absolutely not.

RemieRichards commented on code/modules/clothing/under/ties.dm:
```diff
@@ -367,3 +367,37 @@
+		current_lawyer = null
+
+/obj/item/clothing/tie/lawyers_badge/process()
+	var/obj/item/clothing/under/U = loc
```
No, absolutely not.

Incoming5643 replied:
99% of the time it will short out right below this line, is it really so costly? Accessory code isn't exactly amazing here.

Incoming5643 committed:
removes the process() requirement.
I wonder if remie calls me out on this stuff more than most people because she knows I'll actually fix it.

WJohn commented:
Hold on I've gone and improved the sprite a bit since then. Would you mind using this version before I end up replacing it myself not a week later when I do over the bureaucracy dmi?
![](http://i.imgur.com/VIH3mlA.png)

Incoming5643 committed:
Sprite and Code tweaks

Incoming5643 commented:
You got it

Incoming5643 committed:
Merge branch 'master' of https://github.com/tgstation/-tg-station into hold_it

# Conflicts:
#	code/modules/clothing/under/ties.dm

Incoming5643 commented:
@RemieRichards Removed the process() requirement a few days ago

Incoming5643 commented:
Build fails but not due to this. Master is not compiling due to competing changes

ChangelingRain commented on code/modules/clothing/under/ties.dm:
```diff
@@ -367,3 +371,35 @@
+/obj/item/clothing/tie/lawyers_badge/on_uniform_dropped(obj/item/clothing/under/U)
+	if(!isliving(U.loc))
+		return
+	var/mob/living/L = U.loc
```
You're missing a line, here.

Incoming5643 replied:
Good catch. Somehow the terminal line was lost in the merge resolution. Been fixed.

Incoming5643 committed:
Line ending memes nearly doomed us

ChangelingRain commented:
I hope this handles burning off and suchlike, hmmmmmmm

Incoming5643 commented:
Items don't traditionally burn from the inventory of a person. There's a few tricky ways to get a burning item in your hand but doing ANYTHING to it invites burn damage and drops it. In any case a burning uniform should be handled alright.

 A burning accessory that managed to be attached to a nonburning uniform might not be, but again this is a situation we don't actually allow.
